wytch/view.py

```diff
@@ -700,38 +700,60 @@ class TextInput(ValueWidget):
     def __init__(self, default = "", length = 12, onchange = lambda w, v: None,
             password = False):
         super(TextInput, self).__init__(value = default, onchange = onchange)
-        # TODO: Support longer strings
         self.length = length
         self.onchange = onchange
         self.value = default
+        self.offset = 0
         self.cursor = len(self.value)
         self.password = password
         self.vstretch = False
         self.handlers.extend([
             ("\x7f", self._onbackspace),
+            ("<delete>", self._ondelete),
             ("<left>", self._onleft),
-            ("<right>", self._onright)])
+            ("<right>", self._onright),
+            ("<home>", self._onhome),
+            ("<end>", self._onend)])
 
     def _onbackspace(self, kc):
```

Comment:
This method needs to update self.offset too, otherwise this:
![jklbug](https://cloud.githubusercontent.com/assets/3966931/10834522/52d9f524-7e9a-11e5-8d0c-ba50c2eb8d72.png)

can happen